user-guide/Advanced_Modules/User Defined APIs/Triggering_an_API.md

```diff
@@ -0,0 +1,206 @@
+
+For some of these errors, more information will also be logged in the UserDefinableApiEndpoint.txt logfile. The location of this logfile is configurable, more info [here](xref:UD_APIs_UserDefinableApiEndpoint#logging).
+
+##### FaultingNode
+
+With some errors, a `faultingNode` will be passed. This is the DataMiner ID of the agent that executed the trigger. This can be used to know on what agent the logging should be checked if something went wrong.
+
+##### MissingScriptParameters
+
+If an error with reason `MissingScriptParameters` is returned, this array will contain the names of the missing script parameters.
+
+## Client Test Tool
+
+It is possible to trigger an API through the User Definable API UI in the Client Test Tool. It can be found under `Advanced > Apps > User Definable APIs...` on the `Trigger` tab.
+
+![Client Test Tool Trigger Screenshot](~/user-guide/images/UDAPIS_ClientTestToolTrigger.jpg)
+
+To trigger an API, simply select the API Definition in the dropdown. This will automatically fill in the route in the textbox below. Choose the `RequestMethod` from the dropdown and enter your secret in the Secret textbox. Optionally, you can fill in a request body in the large textbox on the bottom. When clicking 'Trigger', your request is sent and the response will appear along with some stats and info about your request. This is a great way to test your API while developing it, as it is easy to trigger.
+
+## SLNet Message
+
+You can trigger a user defined API by sending the `TriggerUserDefinableApiRequestMessage` SLNet message. This message returns a `TriggerUserDefinableApiResponseMessage`. This way could be useful for easily testing the API without having to send the actual HTTP request.
+
+### TriggerUserDefinableApiRequestMessage
+
+The request message contains the following properties:
+
+|Property     |Type         |Explanation|
+|-------------|-------------|-----------|
+|RequestMethod|RequestMethod|Contains the HTTP request method. See [defining an API](xref:UD_APIs_Define_API#requestmethod).|
+|Route        |string       |The route where the API is available on.|
+|Secret       |string       |The secret that is needed to access the API. See [ApiToken](xref:UD_APIs_Objects_ApiToken#secret).|
+|RawBody      |string       |Contains the request body as a string.|
+
+### TriggerUserDefinableApiResponseMessage
+
+Sending the `TriggerUserDefinableApiRequestMessage` returns the `TriggerUserDefinableApiResponseMessage`. This message contains the following properties:
+
+|Property    |Type          |Explanation|
+|------------|--------------|-----------|
+|TraceData   |TraceData     |Contains `UserDefinableApiTriggerErrorData` in case something goes wrong. It can contain one of the following [reasons](#error-reasons).|
+|ActionResult|IActionResult |Contains an `ActionResult`, for now, this will always be an [AutomationScriptActionResult](#automationscriptactionresult)|
```

Comment:
```suggestion
|ActionResult|IActionResult |Contains an `ActionResult`, for now, this will always be an [AutomationScriptActionResult](#automationscriptactionresult).|
```